Software/src/LogWriter.cpp

```diff
@@ -29,33 +29,42 @@ int LogWriter::initWith(const QString& logsDirPath)
 
 	QDir logsDir(logsDirPath);
 	if (logsDir.exists() == false) {
-		cout << "mkdir " << logsDirPath.toStdString() << endl;
+		std::cout << "mkdir " << logsDirPath.toStdString() << std::endl;
 		if (logsDir.mkdir(logsDirPath) == false) {
-			cerr << "Failed mkdir '" << logsDirPath.toStdString() << "' for logs. Exit." << endl;
+			std::cerr << "Failed mkdir '" << logsDirPath.toStdString() << "' for logs. Exit." << std::endl;
 			return LightpackApplication::LogsDirecroryCreationFail_ErrorCode;
 		}
 	}
 
 	if (rotateLogFiles(logsDir) == false)
-		cerr << "Failed to rotate old log files." << endl;
+		std::cerr << "Failed to rotate old log files." << std::endl;
 
 	const QString logFilePath = logsDirPath + "/Prismatik.0.log";
 	QScopedPointer<QFile> logFile(new QFile(logFilePath));
 	if (logFile->open(QIODevice::WriteOnly | QIODevice::Truncate | QIODevice::Text)) {
 		QMutexLocker locker(&m_mutex);
 
 		m_logStream.setDevice(logFile.take());
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+		m_logStream << Qt::endl;
+#else
 		m_logStream << endl;
+#endif
 
 		const QDateTime currentDateTime(QDateTime::currentDateTime());
 		m_logStream << currentDateTime.date().toString("yyyy_MM_dd") << " ";
-		m_logStream << currentDateTime.time().toString("hh:mm:ss:zzz") << " Prismatik " << VERSION_STR << endl;
+		m_logStream << currentDateTime.time().toString("hh:mm:ss:zzz") << " Prismatik " << VERSION_STR;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+		m_logStream << Qt::endl;
+#else
+		m_logStream << endl;
+#endif
```

Comment:
Urgh.

Reply:
Not blaming anyone, but that's messy

Reply:
For sure.. well when 5.12 is gonna be dropped.. in 2022.. it'll be better :p